### PR TITLE
Clinic database

### PR DESCRIPTION
**Pull request by Tobby8629**
Create Database based on diagram
- create a database ( clinic_records)
- create tables ( patients,  medical_histories, invoices, invoice_items and treaments),
- create the relationships between the table (one to one relationsip and one to many relationship) based on the diagram
- create a joint table between treatments and medical_histories tables to implement the many to many relationship
- create indexing for all foreign keys.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    id BIGINT PRIMARY KEY,
+    unit_price DECIMAL,
+    quantity INT,
+    total_price DECIMAL
+);
+
+--Alter medical histories table
+ALTER TABLE medical_histories
+ADD CONSTRAINT FK_patients_medical_histories
+FOREIGN KEY (patient_id)
```
Comment (Davidosky007) [inline schema_based_on_diagram.sql:45]: * Kindly create indexes for all foreign keys [as stated in the requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md#:~:text=Remember%20to%20add%20the%20FK%20indexes)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Tobby8629 @Thoth1111 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Thoth1111) [reply schema_based_on_diagram.sql:45]: Thank you @Davidosky007. 
I've added the missing index now.

**Review by franclobo — APPROVED** [review]
Hi @Tobby8629, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Correct schema ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**